### ISSUE TITLE
Combining GNN and CNN

### ISSUE THREAD

**Issue by tharindurmt** [rusty1s/pytorch_geometric#1715]
I am looking at how pre-trained using a CNN can be incorporate in to a GNN for further training. The idea is as follows.

Assume that we a have a pre-trained CNN trained on super-pixelated images.  We can "group" the CNN features corresponding to a given super-pixel and a graph can be created. These graphs are then used to train a GNN and the CNN in an end-to-end fashion. 
Although I did a bit of research, I couldn't figure out any way of doing this.

Any suggestions are welcome!

**Comment by rusty1s** [comment]
The idea would be to "scatter" pixel feature information of a superpixel into a single representation. Consider that you have a "mask" `idx` holding indices to which superpixel a pixel belongs. Then you can do the following to train your model in an end-to-end fashion:
```
idx = ...  # [height, width] holding indices from 0 to N - 1 superixels
img = ... # [height, width, num_features]

node_x = scatter(img.view(-1, num_features), idx.view(-1), dim=0, dim_size=N, reduce="mean")
```

**Comment by tharindurmt** [comment]
@rusty1s  Thank you very much for the insight. I will look into this.

**Comment by DuneDrive** [comment]
@sivaji123256 I was wondering the same thing. Did you also find other resources for combining CNN and GNN?
